internal/controller/dragonfly_instance.go

```diff
@@ -487,12 +531,22 @@ func (dfi *DragonflyInstance) replicaOfNoOne(ctx context.Context, pod *corev1.Po
 		}
 	}
 
-	dfi.log.Info("marking pod role as master", "pod", pod.Name)
-	patchFrom := client.MergeFrom(pod.DeepCopy())
+	masterIp := pod.Status.PodIP
+
+	dfi.log.Info("Marking pod role as master", "pod", pod.Name, "masterIp", masterIp)
 	pod.Labels[resources.RoleLabelKey] = resources.Master
-	delete(pod.Labels, resources.MasterIpLabelKey)
-	if err := dfi.client.Patch(ctx, pod, patchFrom); err != nil {
-		return fmt.Errorf("failed to update the role label on the pod: %w", err)
+	// for compatibily, to be remove in futur version
+	if !strings.Contains(masterIp, ":") {
+		pod.Labels[resources.MasterIpLabel] = masterIp
```

Comment:
please add the label irrespective of its IP type. We will remove it in future. Until then the label should exist. Also you are resetting masterIP here. But in replTakeover, you are not setting masterIP. Please maintain same behaviour across.